src/mlpack/methods/linear_regression/linear_regression_main.cpp

```diff
@@ -19,6 +19,11 @@
 
 #include <mlpack/core/util/mlpack_main.hpp>
 
+#ifdef BINDING_NAME
+  #undef BINDING_NAME
+#endif
+#define BINDING_NAME linear_regression
```

Comment:
This is already done above---is it necessary here?  I think it won't do anything. :+1:

Reply:
In mlpack_main.hpp I undefined BINDING_NAME because all persistent params have to be stored under `""`, so I guess BINDING_NAME should be redefined over here (after importing mlpack_main.hpp)

Reply:
Ah, ok.  Maybe it would be better to just, in `mlpack_main.hpp`, do something like this?

```
#ifdef BINDING_NAME
  #define OLD_BINDING_NAME BINDING_NAME
  #undef BINDING_NAME
#endif
#define BINDING_NAME

// Now define parameters like `verbose` and others...

#ifdef OLD_BINDING_NAME
  #undef BINDING_NAME
  #define BINDING_NAME OLD_BINDING_NAME
  #undef OLD_BINDING_NAME
#endif
```

Let me know what you think.

Reply:
I did this, but I am getting an error. I have pushed the changes, can you take a look?

Reply:
I did some debugging and I found that in the code above after BINDING_NAME is undefined the value of OLD_BINDING_NAME also changes, it then becomes BINDING_NAME literally and is not linear_regression. Similarly, in the second block of code after OLD_BINDING_NAME is undefined then BINDING_NAME becomes OLD_BINDING_NAME literally and is not linear_regression. Due to this, all the parameters are registered under the name OLD_BINDING_NAME and I am getting an error related to unknown parameters. It is just weird that the BINDING_NAME is not just replaced by linear_regression while defining OLD_BINDING_NAME in the first place. 
I think that this is the reason for the error, let me know what you think?

Reply:
Did some more research and found that their is no direct way of "saving" a `#define`'s value and the redefining to the saved value. There is however `#pragma push_macro` and `#pragma pop_macro` which are generally used for such tasks but these are not standard and might not be available in every compiler. I guess we would need to think of something else over here.